tests/integration/test_schema.py

```diff
@@ -1813,3 +1813,154 @@ async def test_invalid_schemas(registry_async_client):
+    res = await registry_async_client.delete("subjects/{}/versions/1?permanent=true".format(subject))
+    assert res.status_code == 200
+
+    # Cannot hard delete twice
+    res = await registry_async_client.delete("subjects/{}/versions/1?permanent=true".format(subject))
+    assert res.status_code == 404
+    assert res.json()["error_code"] == 40402
+    assert res.json()["message"] == "Version 1 not found."
+
+
+async def test_schema_hard_delete_whole_schema(registry_async_client):
+    subject = new_random_name("subject")
+    res = await registry_async_client.put("config", json={"compatibility": "BACKWARD"})
+    assert res.status == 200
+    schemav1 = {
+        "type": "record",
+        "name": "myenumtest",
+        "fields": [{
+            "type": {
+                "type": "enum",
+                "name": "enumtest",
+                "symbols": ["first", "second"],
+            },
+            "name": "faa",
+        }]
+    }
+    res = await registry_async_client.post(
+        f"subjects/{subject}/versions",
+        json={"schema": jsonlib.dumps(schemav1)},
+    )
+    assert res.status == 200
+    assert "id" in res.json()
+    schemav1_id = res.json()["id"]
+
+    schemav2 = {
+        "type": "record",
+        "name": "myenumtest",
+        "fields": [{
+            "type": {
+                "type": "enum",
+                "name": "enumtest",
+                "symbols": ["first", "second", "third"],
+            },
+            "name": "faa",
+        }]
+    }
+    res = await registry_async_client.post(
+        f"subjects/{subject}/versions",
+        json={"schema": jsonlib.dumps(schemav2)},
+    )
+    assert res.status == 200
+    assert "id" in res.json()
+    schemav2_id = res.json()["id"]
+    assert schemav1_id != schemav2_id
+
+    # Hard delete whole schema cannot be done before soft delete
+    res = await registry_async_client.delete("subjects/{}?permanent=true".format(subject))
+    assert res.status_code == 404
+    assert res.json()["error_code"] == 40405
+    assert res.json()["message"] == f"Subject '{subject}' was not deleted first before being permanently deleted"
+
+    # Soft delete whole schema
+    res = await registry_async_client.delete("subjects/{}".format(subject))
+    assert res.status_code == 200
+    assert res.json() == [1, 2]
+
+    res = await registry_async_client.get("subjects/{}/versions".format(subject))
+    assert res.status_code == 404
+    assert res.json()["error_code"] == 40401
+    assert res.json()["message"] == f"Subject '{subject}' not found."
+
+    # Hard delete whole schema
+    res = await registry_async_client.delete("subjects/{}?permanent=true".format(subject))
+    assert res.status_code == 200
+    assert res.json() == [1, 2]
+
+    res = await registry_async_client.get("subjects/{}/versions".format(subject))
+    assert res.status_code == 404
+    assert res.json()["error_code"] == 40401
+    assert res.json()["message"] == f"Subject '{subject}' not found."
```

Comment:
Please add code to create a new schema for the deleted subject. I don't know what should be the behavior then, if the IDs are reset or they continue from the previous value.

Reply:
We should also double check if there is such a test for soft deletes